plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/ModulesManager.java

```diff
@@ -635,6 +641,20 @@ public static void buildKeysForRegularEntries(IProgressMonitor monitor, ModulesF
                 }
             }
         }
+
+        for (Map.Entry<String, File> packageEntry : packages.entrySet()) {
+            buffer.clear();
+
+            File f = packageEntry.getValue();
+
+            ModulesKey modulesKeyForFolder = new ModulesKeyForFolder(
+                    buffer.append(packageEntry.getKey()).append(".__init__").toString(), f);
+
+            if (!keys.containsKey(modulesKeyForFolder)
+                    || PythonPathHelper.isValidSourceFile(f.getName(), dottedValidSourceFiles)) {
```

Comment:
I think you don't need to check for `PythonPathHelper.isValidSourceFile(f.getName(), dottedValidSourceFiles)` in this case (before putting on packages it was already pre-checked).